src/utils/GraphQLHints.js

```diff
@@ -0,0 +1,385 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import {
+  isInputType,
+  isCompositeType,
+  isAbstractType,
+  getNullableType,
+  getNamedType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLList,
+  GraphQLBoolean,
+  doTypesOverlap,
+} from 'graphql';
+import {
+  SchemaMetaFieldDef,
+  TypeMetaFieldDef,
+  TypeNameMetaFieldDef,
+} from 'graphql/type/introspection';
+
+import forEachState from './forEachState';
+import hintList from './hintList';
+import objectValues from './objectValues';
+
+import runParser from './runParser';
+import { LexRules, ParseRules, isIgnored } from './Rules';
+
+// Finds all fragment definition ASTs in a source.
+function getFragmentDefinitions(sourceText) {
+  const fragmentDefs = [];
+  runParser(sourceText, {
+    eatWhitespace: stream => stream.eatWhile(isIgnored),
+    LexRules,
+    ParseRules
+  }, (stream, state) => {
+    if (state.kind === 'FragmentDefinition' && state.name && state.type) {
+      fragmentDefs.push({
+        kind: 'FragmentDefinition',
+        name: {
+          kind: 'Name',
+          value: state.name,
+        },
+        typeCondition: {
+          kind: 'NamedType',
+          name: {
+            kind: 'Name',
+            value: state.type,
+          }
+        }
+      });
+    }
+  });
+
+  return fragmentDefs;
+}
+
+export default function GraphQLHints(cur, token, schema, sourceText) {
```

Comment:
This could use a better function name and should be `lowerCamelCase`.

Perhaps `getHintsAtPosition`?

It also deserves to have a doc-block that describes what it does and how to supply the right arguments for cur & token.

IMHO the `schema` and `sourceText` should be the first two arguments as well.